iot_verifier/src/runner.rs

```diff
@@ -458,3 +430,24 @@ fn poc_challengee_reward_unit(num_witnesses: u32) -> anyhow::Result<Decimal> {
     };
     Ok(reward_units.round_dp(SCALING_PRECISION))
 }
+
+/// maybe shuffle & split the verified witness list
+// if the number of witnesses exceeds our cap
+// split the list into two
+// one representing selected witnesses
+// the other representing unselected witnesses
+fn shuffle_and_split_witnesses(
```

Comment:
Can you add a unit test for this please to ensure that the incoming witnesses list is split correctly? There's a `max_witnesses_per_receipt_test` in receipt_txn module in the injector.

Reply:
test added here: e103d9b81792652d7f391e99a42b78d7254e19b5